init-dev-DB.sh

```diff
@@ -11,6 +11,65 @@ else
   echo Looks like PostgreSQL is already installed.
 fi
 
+# ------------------------------------------------------------------------------
+# The next section repairs possible missing folders whose absence prevent
+# Postgres from starting correctly, This problem occurs with the latest
+# versions of macOS (e.g. Yosemite, El Capitan and Sierra).
+
+if [ ! -d "/usr/local/var/postgres/pg_tblspc" ]; then
+    mkdir /usr/local/var/postgres/pg_tblspc
+fi
+
+if [ ! -d "/usr/local/var/postgres/pg_twophase" ]; then
+    mkdir /usr/local/var/postgres/pg_twophase
+fi
+
+if [ ! -d "/usr/local/var/postgres/pg_stat" ]; then
+    mkdir /usr/local/var/postgres/pg_stat
+fi
+
+if [ ! -d "/usr/local/var/postgres/pg_stat_tmp" ]; then
+    mkdir /usr/local/var/postgres/pg_stat_tmp
+fi
+
+if [ ! -d "/usr/local/var/postgres/pg_replslot" ]; then
+    mkdir /usr/local/var/postgres/pg_replslot
+fi
+
+if [ ! -d "/usr/local/var/postgres/pg_snapshots" ]; then
+    mkdir /usr/local/var/postgres/pg_snapshots
+fi
+
+if [ ! -d "/usr/local/var/postgres/pg_stat_tmp" ]; then
+    mkdir /usr/local/var/postgres/pg_stat_tmp
+fi
+
+if [ ! -d "/usr/local/var/postgres/pg_replslot" ]; then
+    mkdir /usr/local/var/postgres/pg_replslot
+fi
+
+if [ ! -d "/usr/local/var/postgres/pg_snapshots" ]; then
+    mkdir /usr/local/var/postgres/pg_snapshots
+fi
+
+if [ ! -d "/usr/local/var/postgres/pg_commit_ts" ]; then
+    mkdir /usr/local/var/postgres/pg_commit_ts
+fi
+
+if [ ! -d "/usr/local/var/postgres/pg_logical" ]; then
+    mkdir /usr/local/var/postgres/pg_logical
+fi
+
+if [ ! -d "/usr/local/var/postgres/pg_logical/snapshots" ]; then
+    mkdir /usr/local/var/postgres/pg_logical/snapshots
+fi
+
+if [ ! -d "/usr/local/var/postgres/pg_logical/mappings" ]; then
+    mkdir /usr/local/var/postgres/pg_logical/mappings
+fi
+
```

Comment:
Happy with the logic of the loop but I think we still need to perform a file check because some of those folders might actually already exist [1] (and `mkdir` complains when we attempt to create an existing folder).

[1] Some of then already existed for me